api/document/serializers/content.py

```diff
@@ -123,19 +127,34 @@ def get_content_type(self, instance: NestableAnnotation):
         return self.CONTENT_TYPE
 
     def get_inlines(self, instance: NestableAnnotation):
+        if self.IS_LEAF:
+            return []
         return NestedAnnotationSerializer(
             instance.children, context=self.context, many=True).data
 
     def get_text(self, instance: Annotation):
         return self.doc_node_text[instance.start:instance.end]
 
+    def to_internal_value(self, data: JsonDict) -> JsonDict:
+        inlines = [] if self.IS_LEAF else data.get('inlines', [])
+        result = super().to_internal_value(data)
+        result['content_type'] = self.CONTENT_TYPE
+        result['inlines'] = []
+        if inlines:
+            serializer = NestedAnnotationSerializer()
+            result['inlines'].extend([
+                serializer.to_internal_value(item) for item in inlines
+            ])
```

Comment:
I wonder if this makes more sense as a custom field like the `children` field of DocNodes? It feels like we're fighting DRF here.

Reply:
Yeah, as I think I probably [mentioned while you were reviewing](https://github.com/18F/omb-eregs/pull/815#discussion_r160726146), this part of the code was done rather hastily, partly because I was getting really tired of making custom field types and just wanted to finish up the PR. As I mentioned to you in Slack, I probably should have reduced the scope of this PR but hid the `PUT` endpoint behind a feature flag so that I wouldn't be worried about the incomplete functionality being abused.